version: "3.3"

services:
  django:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    # command: gunicorn app.wsgi:application --bind 0.0.0.0:8000
    ports:
      - "${APP_PORT:-8080}:8000"
    volumes:
      - .:/home/app/web
      - static_volume:/home/app/static
    networks:
      - fnf-django
  nginx:
    build: ./nginx
    ports:
      - "${NGINX_PORT:-80}:80"
    depends_on:
      - django
    links:
      - django
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/conf.d/default.conf
      - static_volume:/home/app/static
    networks:
      - fnf-django

  db:
    image: postgres:14.0-alpine
    restart: always
    cap_add:
      - SYS_NICE
    environment:
      PGDATA: /data/postgres
      POSTGRES_USER: admin
      POSTGRES_DB: admin
      POSTGRES_PASSWORD: "POSzolGy7o9nCdfBaiNDSGTQQaRERLaQ"
    volumes:
      - "./data:/data/postgres"
      - ./postgresql.conf:/var/lib/postgresql/data/postgresql.conf
    ports:
      - "5432:5432"
    networks:
      - fnf-django

volumes:
  static_volume:

networks:
  fnf-django:
    driver: bridge
